openedx/core/djangoapps/content/course_overviews/tests.py

```diff
@@ -0,0 +1,152 @@
+            'clean_id',
+            'has_ended',
+            'has_started',
+            'start_datetime_text',
+            'end_datetime_text',
+        ]
+        for method_name in methods_to_test:
+            course_value = getattr(course, method_name)()
+            cache_miss_value = getattr(course_overview_cache_miss, method_name)()
+            cache_hit_value = getattr(course_overview_cache_hit, method_name)()
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+        # Other values to test
+        others_to_test = [(
+            course_image_url(course),
+            course_overview_cache_miss.course_image_url,
+            course_overview_cache_hit.course_image_url
+        )]
+        for (course_value, cache_miss_value, cache_hit_value) in others_to_test:
+            self.assertEqual(course_value, cache_miss_value)
+            self.assertEqual(cache_miss_value, cache_hit_value)
+
+    @ddt.data(*itertools.product(
+        [
+            {
+                "static_asset_path": "/my/cool/path",
+                "display_name": "Test Course",
+                "start": YESTERDAY,
+                "end": TOMORROW,
+            },
+            {}
+        ],
+        [ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split]
+    ))
+    @ddt.unpack
+    def test_course_overview_behavior(self, course_kwargs, modulestore_type):
+        """Tests if CourseOverviews and CourseDescriptors behave the same
+        by comparing pairs of them given a variety of scenarios.
+
+        Args:
+            course_kwargs (dict): kwargs to be passed to course constructor
+            modulestore_type (ModuleStoreEnum.Type)
+            is_user_enrolled (bool)
+        """
+
+        course = CourseFactory.create(
+            course="test_course",
+            org="edX",
+            default_store=modulestore_type,
+            **course_kwargs
+        )
+        self.check_course_overview_against_course(course)
+
+    def test_course_overview_cache_invalidation(self):
+        """
+        Tests that when a course is published, the corresponding
+        course_overview is removed from the cache.
+        """
+
+        # Create a course where mobile_available is True.
+        course = CourseFactory.create(
+            course="test_course",
+            org="edX",
+            mobile_available=True
+        )
+        course_overview1 = CourseOverview.get_from_id(course.id)
+        self.assertTrue(course_overview1.mobile_available)
+
+        # Set mobile_available to False and update the course.
+        # This fires a course_published signal, which should be caught in signals.py, which should in turn
+        # delete the corresponding CourseOverview from the cache.
+        course.mobile_available = False
+        self.store.update_item(course, ModuleStoreEnum.UserID.test)
+        time.sleep(0.01)  # Give a few milliseconds for the delete query to take place
```

Comment:
Is this `sleep` needed?  Or does all the signal handling happen synchronously in test code?
If the `sleep` is needed, this is vulnerable to becoming a flaky test on Jenkins.